sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -510,32 +539,93 @@ class Analyzer(
+    // database with that the referenced view has, so we need to use the variable `defaultDatabase`
+    // to track the current default database.
+    // When the relation we resolve is a view, we fetch the view.desc(which is a CatalogTable), and
+    // then set the value of `CatalogTable.viewDefaultDatabase` to the variable `defaultDatabase`,
+    // we look up the relations that the view references using the default database.
+    // For example:
+    // |- view1 (defaultDatabase = db1)
+    //   |- operator
+    //     |- table2 (defaultDatabase = db1)
+    //     |- view2 (defaultDatabase = db2)
+    //        |- view3 (defaultDatabase = db3)
+    //   |- view4 (defaultDatabase = db4)
+    // In this case, the view `view1` is a nested view, it directly references `table2`、`view2`
+    // and `view4`, the view `view2` references `view3`. On resolving the table, we look up the
+    // relations `table2`、`view2`、`view4` using the default database `db1`, and look up the
+    // relation `view3` using the default database `db2`.
+    //
+    // Note this is compatible with the views defined by older versions of Spark(before 2.2), which
+    // have empty defaultDatabase and all the relations in viewText have database part defined.
+    def resolveRelation(
+        plan: LogicalPlan,
+        defaultDatabase: Option[String] = None): LogicalPlan = plan match {
+      case u @ UnresolvedRelation(table: TableIdentifier, _) if isRunningDirectlyOnFiles(table) =>
+        u
+      case u: UnresolvedRelation =>
+        val defaultDatabase = AnalysisContext.get.defaultDatabase
+        val relation = lookupTableFromCatalog(u, defaultDatabase)
+        resolveRelation(relation, defaultDatabase)
+      // Hive support is required to resolve a persistent view, the logical plan returned by
+      // catalog.lookupRelation() should be:
+      // `SubqueryAlias(_, View(desc: CatalogTable, desc.output, child: LogicalPlan), _)`,
+      // where the child should be a logical plan parsed from `desc.viewText`.
+      // If the child of a view is empty, we will throw an AnalysisException later in
+      // `checkAnalysis`.
+      case view @ View(desc, _, Some(child)) =>
+        val context = AnalysisContext(defaultDatabase = desc.viewDefaultDatabase)
+        // Resolve all the UnresolvedRelations and Views in the child.
+        val newChild = AnalysisContext.withAnalysisContext(context) {
+          execute(child)
+        }
+        view.copy(child = Some(newChild))
+      case p @ SubqueryAlias(_, view: View, _) =>
+        val newChild = resolveRelation(view, defaultDatabase)
+        p.copy(child = newChild)
+      case _ => plan
+    }
+
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+      case i @ InsertIntoTable(u: UnresolvedRelation, parts, child, _, _) if child.resolved =>
+        i.copy(table = EliminateSubqueryAliases(lookupTableFromCatalog(u)))
```

Comment:
Should we check that we are not in a nested context? I suppose that CreateView already does something similar?